### PR TITLE
Retention period updates

### PR DESCRIPTION
**Pull request by aarohib**
## What does this PR do?
- Updates retention period details. [Preview](https://deploy-preview-1649--mystifying-fermi-1bc096.netlify.app/concepts/what-is-a-retention-period/)
- Adds registering Namespaces using SDKs to the app-dev guide. [Preview](https://deploy-preview-1649--mystifying-fermi-1bc096.netlify.app/application-development/features#namespaces)
- Updates the Namespaces concept. [Preview](https://deploy-preview-1649--mystifying-fermi-1bc096.netlify.app/namespaces)

### CODE REVIEW

File: assembly/guide-configs/app-dev/features.json [hunk 1]
```diff
@@ -609,6 +609,31 @@
...
+        },
+        {
+          "lang": "java",
+          "id": "java/how-to-register-namespace-in-java"
+        },
+        {
+          "lang": "php",
+          "id": "none"
+        },
+        {
```
Comment (flossypurse) [inline assembly/guide-configs/app-dev/features.json:632]: missing python here

File: docs-src/app-dev-context/namespaces.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  - guide-context
+---
+
+A Namespace is a unit of isolation within the Temporal Platform.
+
+You can use Namespaces to match the development lifecycle; for example, having separate `dev` and `prod` Namespaces.
+Or you could use them to ensure Workflow Executions between different teams never communicate; such as ensuring that the `teamA` Namespace never impacts the `teamB` Namespace.
+
+The register and update Namespace APIs send a gRPC request to the Frontend Service to create or update a Namespace.
+Use a custom authorizor on your Frontend Service in the Temporal Cluster to set restrictions on who can create or update Namespaces.
```
Comment (flossypurse) [inline docs-src/app-dev-context/namespaces.md:16]: I think this is the same Authorizer Plugin we have defined in terms?
If so - let us link to it.

File: docs/application-development/features.md [hunk 3]
```diff
@@ -3182,3 +3183,67 @@ async function yourWorkflow() {
 
 </TabItem>
 </Tabs>
+
+## Namespaces
```
Comment (flossypurse) [inline docs/application-development/features.md:3187]: Hey this is pretty great. Good thinking on adding this to the guide.

File: docs/concepts/what-is-a-retention-period.md [hunk 4]
```diff
@@ -8,13 +8,19 @@ tags:
   - explanation
 ---
 
-A Retention Period is the amount of time a Workflow Execution Event History remains in the Cluster's persistence store.
+A Retention Period is the amount of time, in days, that a Workflow Execution Event History remains in the Cluster's persistence store.
 
 - [How to set the Retention Period for the Namespace](/tctl/namespace/register/#--retention)
+- [How to set the Retention Period for the Namespace using SDK](/application-development/features/#namespaces)
```
Comment (flossypurse) [inline docs/concepts/what-is-a-retention-period.md:14]:
```suggestion
- [How to set the Retention Period for a Namespace using an SDK](/application-development/features/#namespaces)
```

You should point this link to the app dev context src id - linkMagic will then rewrite it.

File: docs/concepts/what-is-a-retention-period.md [hunk 5]
```diff
@@ -8,13 +8,19 @@ tags:
...
 
 A Retention Period applies to a single [Namespace](/concepts/what-is-a-namespace) and is set when the Namespace is registered.
 
-If the Retention Period isn't set, it defaults to 2 days.
+The Temporal Cluster triggers a Timer task at the end of the Retention Period that cleans up the associated Workflow Execution Event History on that Namespace.
```
Comment (flossypurse) [inline docs/concepts/what-is-a-retention-period.md:18]: All of this is great stuff.

File: docs/java/how-to-register-namespace-in-java.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
+---
+id: how-to-register-namespace-in-java
```
Comment (flossypurse) [inline docs/java/how-to-register-namespace-in-java.md:2]: @rachfop any chance we could get Namespace creation examples for the other languages?

**Review by flossypurse — COMMENTED** [review]
Ya, I dig this. Great work getting the details and the defaults - it really adds a lot of value here.

File: assembly/guide-configs/app-dev/features.json (hunk 1 above)
Comment (aarohib) [reply assembly/guide-configs/app-dev/features.json:632]: updated

File: docs-src/app-dev-context/namespaces.md (hunk 2 above)
Comment (aarohib) [reply docs-src/app-dev-context/namespaces.md:16]: Yes! linked

File: docs/java/how-to-register-namespace-in-java.md (hunk 6 above)
Comment (aarohib) [reply docs/java/how-to-register-namespace-in-java.md:2]: I'm adding the topic for Go.

File: docs/concepts/what-is-a-retention-period.md [hunk 7]
```diff
@@ -8,13 +8,19 @@ tags:
...
+- [How to set the Retention Period for a Namespace](/tctl/namespace/register/#--retention)
+- [How to set the Retention Period for a Namespace using an SDK](/application-development/features/#namespaces)
 
 A Retention Period applies to a single [Namespace](/concepts/what-is-a-namespace) and is set when the Namespace is registered.
 
-If the Retention Period isn't set, it defaults to 2 days.
+The Temporal Cluster triggers a Timer task at the end of the Retention Period that cleans up the associated Workflow Execution Event History on that Namespace.
+
+On Temporal Cluster version 1.18 and later, Namespaces can use any Retention Period as needed. Ensure that your Persistence store has enough capacity for the storage.
+The maximum retention period on Temporal Cluster versions 1.17 and before is 30 days.
```
Comment (aarohib) [inline docs/concepts/what-is-a-retention-period.md:21]:
```suggestion
The maximum retention period on Temporal Cluster versions 1.17 and earlier is 30 days.
```

File: docs/clusters.md [hunk 8]
```diff
@@ -276,15 +276,21 @@ Ports are configurable in the Cluster configuration.
...
 ### Retention Period
 
-A Retention Period is the amount of time a Workflow Execution Event History remains in the Cluster's persistence store.
+A Retention Period is the amount of time, in days, that a Workflow Execution Event History remains in the Cluster's persistence store.
 
-- [How to set the Retention Period for the Namespace](/tctl/namespace/register/#--retention)
+- [How to set the Retention Period for a Namespace](/tctl/namespace/register/#--retention)
+- [How to set the Retention Period for a Namespace using an SDK](/application-development/features/#namespaces)
 
 A Retention Period applies to a single [Namespace](/namespaces#) and is set when the Namespace is registered.
```
Comment (yiminc) [inline docs/clusters.md:284]: It is weird to say that it applies to a 'single' Namespace. I would say it applies to all closed workflows within that namespace.

File: docs/clusters.md [hunk 9]
```diff
@@ -276,15 +276,21 @@ Ports are configurable in the Cluster configuration.
...
 
-If the Retention Period isn't set, it defaults to 2 days.
+The Temporal Cluster triggers a Timer task at the end of the Retention Period that cleans up the associated Workflow Execution Event History on that Namespace.
+
+On Temporal Cluster version 1.18 and later, Namespaces can use any Retention Period as needed. Ensure that your Persistence store has enough capacity for the storage.
```
Comment (yiminc) [inline docs/clusters.md:288]: There is still minimal retention requirement.

File: docs/concepts/what-is-a-namespace.md [hunk 10]
```diff
@@ -10,23 +10,54 @@ tags:
...
+
+Namespaces are created on the Temporal Cluster, and provide a range of controls to achieve isolation on Workflow Executions.
+
+- You can use Namespaces to match the development lifecycle; for example, having separate `dev` and `prod` Namespaces.
 - If no other Namespace is specified, the Temporal Cluster uses the Namespace "default" for all Temporal SDKs and tctl.
-  - If you are deploying through Docker Compose or using the [auto-setup image](https://github.com/temporalio/docker-builds/blob/main/docker/auto-setup.sh) in a custom Docker Compose application, the Namespace "default" is created, through the auto-setup script.
-  - If you are deploying through the [Temporal Helm charts](https://github.com/temporalio/helm-charts), you can create the "default" Namespace by using tctl; for example, `tctl namespace default`.
-    We recommend using the default Namespace if you aren’t using multiple Namespaces.
+  See the [Registration](#registration) section for details.
 - **Case Insensitive**: Because of DNS, Namespaces are case insensitive on the network and routing side.
```
Comment (yiminc) [inline docs/concepts/what-is-a-namespace.md:22]: This is not correct.

In OSS, namespace is case sensitive. You could create namespace foo and Foo, and they are 2 different namespaces.

In cloud, they are case insensitive. In fact, cloud only allow you to create lower case namespace names.

File: docs/concepts/what-is-a-namespace.md [hunk 11]
```diff
@@ -10,23 +10,54 @@ tags:
...
+All SDKs require a Namespace on the Cluster for their Client calls. If not set using Client options, the Workflow Client API looks for the `default` Namespace. If there is no default Namespace registered with your Temporal Cluster, all calls will throw errors.
+You must register your Namespace with the Temporal Cluster before setting it in your Client.
+
+You can register your Namespaces in the following ways:
+
+- In your Cluster setup, create your Namespaces, including the default, in your setup script.
+  For example:
+
+  - If deploying through Docker Compose or using the [auto-setup image](https://github.com/temporalio/docker-builds/blob/main/docker/auto-setup.sh) in a custom Docker Compose application, the Namespace "default" is created, through the auto-setup script.
+  - If deploying through the [Temporal Helm charts](https://github.com/temporalio/helm-charts), you can create the "default" Namespace by using tctl; for example, `tctl namespace default`.
```
Comment (yiminc) [inline docs/concepts/what-is-a-namespace.md:44]: I don't think `tctl namespace default` will create you a namespace. Please try it to verify.

File: docs/concepts/what-is-a-namespace.md [hunk 12]
```diff
@@ -10,23 +10,54 @@ tags:
...
+
+- In your Client program, register your Namespace using `RegisterNamespaceRequest` API available in all the SDKs.
+
+  - [How to register a new Namespace using SDK](/application-development/features#namespaces)
+
+- Use the `tctl namespace register` command with the `--retention` modfiier to register your Namespaces, one by one, and set the Retention Period on each.
```
Comment (yiminc) [inline docs/concepts/what-is-a-namespace.md:50]: This should be below next line [How to register a new Namespace using tctl] ?

File: docs/concepts/what-is-a-retention-period.md [hunk 13]
```diff
@@ -8,13 +8,19 @@ tags:
...
+A Retention Period is the amount of time, in days, that a Workflow Execution Event History remains in the Cluster's persistence store.
 
-- [How to set the Retention Period for the Namespace](/tctl/namespace/register/#--retention)
+- [How to set the Retention Period for a Namespace](/tctl/namespace/register/#--retention)
+- [How to set the Retention Period for a Namespace using an SDK](/application-development/features/#namespaces)
 
 A Retention Period applies to a single [Namespace](/concepts/what-is-a-namespace) and is set when the Namespace is registered.
 
-If the Retention Period isn't set, it defaults to 2 days.
+The Temporal Cluster triggers a Timer task at the end of the Retention Period that cleans up the associated Workflow Execution Event History on that Namespace.
```
Comment (yiminc) [inline docs/concepts/what-is-a-retention-period.md:18]: Not only event history, but also mutable state, current run record (if it is still current run), and visibility record. But those maybe too details, so I I would just say that it cleans up data for the associated Workflow Execution.

File: docs-src/go/how-to-register-namespace-in-go.md [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+        })
+```
+
+The Retention Period setting using `WorkflowExecutionRetentionPeriod` is mandatory.
+The minimum value you can set for this period is 1 day.
+
+Once registered, set Namespace using `Dial` in a Workflow Client to run your Workflow Executions within that Namespace.
+See [how to set Namespace in a Client in Go](/app-dev-context/connect-to-a-cluster) for details.
+
+Note that Namespace registration using this API takes up to 15 seconds to complete.
```
[the remark this answers is not in the document]
Comment (aarohib) [reply docs-src/go/how-to-register-namespace-in-go.md:29]: updated

File: docs-src/references/cluster-metrics.md [hunk 15]
```diff
@@ -24,10 +24,9 @@ Temporal emits metrics for each gRPC service request.
 These metrics are emitted with `type`, `operation`, and `namespace` tags, which provide visibility into Service usage and show the request rates across Services, Namespaces, and Operations.
 
 - Use the `operation` tag in your query to get request rates, error rates, or latencies per operation.
-- Use the `service_name` tag with the [service role tag values](https://github.com/temporalio/temporal/blob/bba148cf1e1642fd39fa0174423b183d5fc62d95/common/metrics/defs.go#L108) to get details for the specific service.
+- Use the `service_name` tag with the [service role tag values](https://github.com/temporalio/temporal/blob/bba148cf1e1642fd39fa0174423b183d5fc62d95/common/metrics/defs.go) to get details for the specific service.
```
[the remark this answers is not in the document]
Comment (aarohib) [reply docs-src/references/cluster-metrics.md:27]: added it back in.

File: docs/concepts/what-is-a-namespace.md (hunk 10 above)
Comment (aarohib) [reply docs/concepts/what-is-a-namespace.md:22]: updated this info

File: docs/concepts/what-is-a-namespace.md (hunk 11 above)
Comment (aarohib) [reply docs/concepts/what-is-a-namespace.md:44]: corrected.

File: docs/clusters.md [hunk 16]
```diff
@@ -276,15 +276,21 @@ Ports are configurable in the Cluster configuration.
 
 ### Retention Period
 
-A Retention Period is the amount of time a Workflow Execution Event History remains in the Cluster's persistence store.
+A Retention Period is the amount of time, in days, that a Workflow Execution Event History remains in the Cluster's persistence store.
```
[the remark this answers is not in the document]
Comment (aarohib) [reply docs/clusters.md:279]: updated

File: docs/clusters.md [hunk 17]
```diff
@@ -276,15 +276,21 @@ Ports are configurable in the Cluster configuration.
...
 
-- [How to set the Retention Period for the Namespace](/tctl/namespace/register/#--retention)
+- [How to set the Retention Period for a Namespace](/tctl/namespace/register/#--retention)
+- [How to set the Retention Period for a Namespace using an SDK](/application-development/features/#namespaces)
 
 A Retention Period applies to a single [Namespace](/namespaces#) and is set when the Namespace is registered.
 
-If the Retention Period isn't set, it defaults to 2 days.
+The Temporal Cluster triggers a Timer task at the end of the Retention Period that cleans up the associated Workflow Execution Event History on that Namespace.
```
[the remark this answers is not in the document]
Comment (aarohib) [reply docs/clusters.md]: We decided to call the group of all Temporal services a Temporal Cluster so using that here to keep it consistent with the rest of docs/messaging.

File: docs-src/app-dev-context/features.md [hunk 18]
```diff
@@ -30,3 +30,4 @@ In this section you can find the following:
 - [How to set Activity timeouts & retries](#activity-timeouts)
 - [How to Heartbeat an Activity](#activity-heartbeats)
 - [How to Asynchronously complete an Activity](#asynchronous-activity-completion)
+- [How to register Namespaces](#namespaces)
```
Comment (cretz) [inline docs-src/app-dev-context/features.md:33]: I am not sure this is how we want to tell people to register namespaces. Yes they _can_ programmatically register namespaces, and yes we may want to document this as a _possibility_, but I think we encourage use of `tctl` here.

Comment (aarohib) [reply docs-src/app-dev-context/features.md:33]: yeah. Updated the description to:
"Use the tctl command-line tool to [register Namespaces](/tctl/namespace/register).
You can also register your Namespaces programmatically with the `RegisterNamespace` API."

Comment (cretz) [reply docs-src/app-dev-context/features.md:33]: :+1: So where does the `/go/how-to-register-namespace-in-go.md` file contents get included in the dev guide? Can we delete that file?

Comment (aarohib) [reply docs-src/app-dev-context/features.md:33]: Yeah; removed it from the docs/go so it should show only in the app dev guide context here: https://deploy-preview-1649--mystifying-fermi-1bc096.netlify.app/application-development/features?lang=go#namespaces

Comment (cretz) [reply docs-src/app-dev-context/features.md:33]: I think programmatically registering a namespace should not be in the app dev guide. While Go unfortunately has a high level wrapper for this, other SDKs consider this a low-level service call like the dozen of others we don't document in the app dev guide.

But this is subjective. I would like to hear others' opinions on its value.

Comment (aarohib) [reply docs-src/app-dev-context/features.md:33]: I've seen a few questions come up about this on Slack and on community, and we point them to the APIs. I get what you're saying though;
If you look at it in the context of "this is a feature (Namespaces), and these are all the things you can do about it (register, update, set)", would the app dev guide inclusion make sense?

Comment (flossypurse) [reply docs-src/app-dev-context/features.md:33]: I think it is fair to call out that Namespace creation is most commonly done from the command line tool -
But from a product coverage perspective, this makes things a lot more complete - and that matters a lot.

Comment (tsurdilo) [reply docs-src/app-dev-context/features.md:33]: I thought all SDKs expose public api (RegisterNamespaceRequest)? 
Yeah we had users ask us "how i register ns with code" in the past cause they dont want to use entry point scripts but if its in docs or not i guess then we should document the entire api? i really don't know

Comment (cretz) [reply docs-src/app-dev-context/features.md:33]: >  But from a product coverage perspective, this makes things a lot more complete - and that matters a lot.

Then I have a lot more low-level API calls for you to document :-) List, update, describe, and delete/deprecate namespace for example. I couldn't imagine us having programmatic sections for all of those though arguably some of those are more useful than the one-time create.

File: docs/concepts/what-is-a-namespace.md [hunk 19]
```diff
@@ -10,23 +10,84 @@ tags:
...
-
 A single Namespace is still multi-tenant.
-You can use Namespaces to match the development lifecycle; for example, having separate `dev` and `prod` Namespaces.
-Or you could use them to ensure Workflow Executions between different teams never communicate; such as ensuring that the `teamA` Namespace never impacts the `teamB` Namespace.
 
+### Usage
+
+Namespaces are created on the Temporal Cluster, and provide a range of controls to achieve isolation on Workflow Executions.
+
+- Namespaces are a mechanism for resource isolation. Heavy traffic from one Namespace will not impact other Namespaces running on the same cluster.
```
Comment (flossypurse) [inline docs/concepts/what-is-a-namespace.md:19]:
```suggestion
- Namespaces are a mechanism for resource isolation. Heavy traffic from one Namespace will not impact other Namespaces running on the same Cluster.
```

**Review by flossypurse — APPROVED** [review]

File: docs/concepts/what-is-a-namespace.md (hunk 19 above)
Comment (aarohib) [reply docs/concepts/what-is-a-namespace.md]: updated

File: docs-src/app-dev-context/features.md (hunk 18 above)
Comment (aarohib) [reply docs-src/app-dev-context/features.md:33]: added high-level info for list, update, describe, and deprecate in concepts for managing Namespaces. Will update the app dev guide with examples for each in another PR.

File: docs/concepts/what-is-a-retention-period.md [hunk 20]
```diff
@@ -8,13 +8,19 @@ tags:
...
+The Temporal Cluster triggers a Timer task at the end of the Retention Period that cleans up the data associated with the closed Workflow Execution on that Namespace.
+mutable
 The minimum Retention Period is 1 day.
-The maximum Retention Period is 30 days.
+On Temporal Cluster version 1.18 and later, the maximum Retention Period value for Namespaces can be set to anything over the minimum requirement of 1 day. Ensure that your Persistence store has enough capacity for the storage.
+On Temporal Cluster versions 1.17 and earlier, the maximum Retention Period you can set is 30 days.
 Setting the Retention Period to 0 results in the error _A valid retention period is not set on request_.
+
+If you don't set the Retention Period value when using the [`tctl namespace register`](/tctl/namespace/register/#--retention) command, it defaults to 3 days.
+If you don't set the Retention Period value when using the [`RegisterNamespaceRequest](/application-development/features/#namespaces) API, it returns an error.
```
Comment (aarohib) [inline docs/concepts/what-is-a-retention-period.md:26]:
```suggestion
If you don't set the Retention Period value when using the [`RegisterNamespaceRequest`](/application-development/features/#namespaces) API, it returns an error.
```

File: docs-src/go/how-to-register-namespace-in-go.md [hunk 21]
```diff
@@ -0,0 +1,36 @@
...
+
+Use [`Register` API](https://pkg.go.dev/go.temporal.io/sdk@v1.17.0/client#NamespaceClient.Register) with the `NamespaceClient` interface to register a [Namespace](/concepts/what-is-a-namespace) and set the [Retention Period](/concepts/what-is-a-retention-period) for the Workflow Execution Event History for the Namespace.
+
+You can also [register Namespaces using the tctl command-line tool](/tctl/namespace/register).
+
+```go
+    client, err := client.NewNamespaceClient(client.Options{HostPort: ts.config.ServiceAddr})
+            //...
+        err = client.Register(ctx, &workflowservice.RegisterNamespaceRequest{
+            Name: your-namespace-name,
```
Comment (tsurdilo) [inline docs-src/go/how-to-register-namespace-in-go.md:20]: This should be Namespace , not Name

File: docs/java/how-to-register-namespace-in-java.md [hunk 22]
```diff
@@ -0,0 +1,38 @@
...
+Use the [`RegisterNamespace` API](https://github.com/temporalio/api/blob/master/temporal/api/workflowservice/v1/service.proto) to register a [Namespace](/concepts/what-is-a-namespace) and set the [Retention Period](/concepts/what-is-a-retention-period) for the Workflow Execution Event History for the Namespace.
+
+```java
+//...
+import io.temporal.api.workflowservice.v1.RegisterNamespaceRequest;
+//...
+public static void createNamespace(String name) {
+    RegisterNamespaceRequest req = RegisterNamespaceRequest.newBuilder()
+            .setNamespace("your-custom-namespace")
+            .setWorkflowExecutionRetentionPeriod(Durations.fromDays(3)) // keeps the Workflow Execution
```
Comment (tsurdilo) [inline docs/java/how-to-register-namespace-in-java.md:21]: Would be nice to also show import to com.google.protobuf.Duration so users know its not java.time

**Comment by tsurdilo** [comment]
lgtm, just left two small comments

File: docs-src/go/how-to-register-namespace-in-go.md (hunk 21 above)
Comment (aarohib) [reply docs-src/go/how-to-register-namespace-in-go.md:20]: updated

**Comment by cretz** [comment]
As an update here, not sure we should 1) document stuff in the app dev guide that cloud users can't do, and 2) have code snippets showing how to do only one (less-frequent-than-others) namespace operation if it's not how we often encourage people to do it.

**Review by yiminc — APPROVED** [review]

**Review by tsurdilo — APPROVED** [review]
thanks for addressing my comments

**Comment by aarohib** [comment]
> As an update here, not sure we should 1) document stuff in the app dev guide that cloud users can't do, and 2) have code snippets showing how to do only one (less-frequent-than-others) namespace operation if it's not how we often encourage people to do it.
@cretz I'll add the other Namespace operations to the app dev guide in a separate PR; Cloud Namespace ops have separate docs which I will link to.

**Comment by cretz** [comment]
> I'll add the other Namespace operations to the app dev guide in a separate PR; Cloud Namespace ops have separate docs which I will link to.

I fear that unless we make it clear in the app dev guide that what we're documenting just doesn't work in what we're hoping will become the primary runtime environment, confusion will occur.

File: docs-src/app-dev-context/namespaces.md [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+sidebar_label: Namespaces
+description: A Namespace is a unit of isolation within the Temporal Platform.
+tags:
+  - guide-context
+---
+
+A Namespace is a unit of isolation within the Temporal Platform.
+
+You can use Namespaces to match the development lifecycle; for example, having separate `dev` and `prod` Namespaces.
+Or you could use them to ensure Workflow Executions between different teams never communicate; such as ensuring that the `teamA` Namespace never impacts the `teamB` Namespace.
```
[the remark this answers is not in the document]
Comment (bergundy) [reply docs-src/app-dev-context/namespaces.md:13]: It's true in most SDKs, we do not expose `namespace` in child workflow options.

File: docs-src/app-dev-context/namespaces.md [hunk 24]
```diff
@@ -0,0 +1,21 @@
...
+---
+
+A [Namespace](/concepts/what-is-a-namespace) is a unit of isolation within the Temporal Platform.
+
+You can use Namespaces to match the development lifecycle; for example, having separate `dev` and `prod` Namespaces.
+Or you could use them to ensure Workflow Executions between different teams never communicate; such as ensuring that the `teamA` Namespace never impacts the `teamB` Namespace.
+
+On Temporal Cloud, use the [Temporal Cloud UI](/cloud-context/namespaces-create) or [tcld commands](https://docs.temporal.io/cloud/tcld/namespace/) to create and manage Namespaces.
+
+On self-hosted Temporal Cluster, you can register and manage your Namespaces using tctl (recommended) or programmatically using APIs. Note that these APIs and tctl commands will not work with Temporal Cloud.
```
Comment (bergundy) [inline docs-src/app-dev-context/namespaces.md:17]: Why is `tctl` recommended?

File: docs-src/references/cluster-metrics.md [hunk 25]
```diff
@@ -26,8 +26,7 @@ These metrics are emitted with `type`, `operation`, and `namespace` tags, which
 - Use the `operation` tag in your query to get request rates, error rates, or latencies per operation.
 - Use the `service_name` tag with the [service role tag values](https://github.com/temporalio/temporal/blob/bba148cf1e1642fd39fa0174423b183d5fc62d95/common/metrics/defs.go#L108) to get details for the specific service.
 
-All common tags that you can add in your query are defined in the following file:
-https://github.com/temporalio/temporal/blob/86966c5d2f78ee74c10b6808857f1a2f64d3c134/common/metrics/defs.go#L90.
+All common tags that you can add in your query are defined in the [defs.go](https://github.com/temporalio/temporal/blob/master/common/metrics/defs.go) file.
```
Comment (bergundy) [inline docs-src/references/cluster-metrics.md:29]: Use a permalink in case this file changes in master.

**Review by bergundy — COMMENTED** [review]
We should really question if adding all of these samples to the app dev guide provide enough value compared to the bloat they add.

I think we would have gotten more value out of documenting the gRPC API and making sure the generated API docs for all languages work well and adding a single section in the app dev guide showing how to call the different gRPC services from the SDK.

**Pull request merged**